libnetdata/socket/socket.c

```diff
@@ -925,11 +925,11 @@ ssize_t netdata_ssl_read(SSL *ssl, void *buf, size_t num) {
 
     int bytes, err, retries = 0;
 
-    //do {
+    do {
         bytes = SSL_read(ssl, buf, (int)num);
         err = SSL_get_error(ssl, bytes);
         retries++;
-    //} while (bytes <= 0 && (err == SSL_ERROR_WANT_READ));
+    } while (err == SSL_ERROR_WANT_READ);
```

Comment:
The way this is now, if the first time `bytes` is positive, the next iteration will overwrite data.

So, if there is possibility to have `SSL_ERROR_WANT_READ` while we have read some data, these data will now be lost (overwritten at like 929 on the second iteration).

I don't know if it is possible to get `SSL_ERROR_WANT_READ` with positive bytes though.

Reply:
wouldn't SSL_MODE_AUTO_RETRY suffice?
in other case should you not wait for bytes on the socket first?

Reply:
`SSL_MODE_AUTO_RETRY` probably yes. I am not familiar with SSL though. Can you make it happen? We need the same on the writing side I think...

Reply:
Yeah, IMHO the corrupted data @ilyam8 mentioned above are just caused by wrong usage of OpenSSL

Reply:
> wouldn't SSL_MODE_AUTO_RETRY suffice? in other case should you not wait for bytes on the socket first?

Timo, we set this on `netdata_ssl_client_ctx`, should we also on `netdata_ssl_srv_ctx` ?

Reply:
I have not checked the code on `srv_ctx` but if you want same behavior (openssl dont bothering you with handling retries ) I suppose yes.